From 9aa9c66be355d95e2c532bbdfdc98bdbba9b6606 Mon Sep 17 00:00:00 2001
From: Michael Simacek <msimacek@redhat.com>
Date: Thu, 7 Dec 2017 15:50:52 +0200
Subject: [PATCH] Support setting "-Xdoclint:none" in m-javadoc-p >= 3.0.0

maven-javadoc-plugin 3.0.0 replaced `additionalparam` string property
with `additionalOptions` list property.
See: https://issues.apache.org/jira/browse/MJAVADOC-368
---
 .../fedoraproject/xmvn/connector/aether/XMvnMojoExecutionListener.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/xmvn-connector-aether/src/main/java/org/fedoraproject/xmvn/connector/aether/XMvnMojoExecutionListener.java b/xmvn-connector-aether/src/main/java/org/fedoraproject/xmvn/connector/aether/XMvnMojoExecutionListener.java
index 5f9b5dd4..bb066638 100644
--- a/xmvn-connector-aether/src/main/java/org/fedoraproject/xmvn/connector/aether/XMvnMojoExecutionListener.java
+++ b/xmvn-connector-aether/src/main/java/org/fedoraproject/xmvn/connector/aether/XMvnMojoExecutionListener.java
@@ -267,7 +267,10 @@ public class XMvnMojoExecutionListener
         // Disable doclint
         if ( JAVADOC_AGGREGATE.equals( execution ) )
         {
+            // maven-javadoc-plugin < 3.0.0
             trySetBeanProperty( mojo, "additionalparam", "-Xdoclint:none" );
+            // maven-javadoc-plugin >= 3.0.0
+            trySetBeanProperty( mojo, "additionalOptions", new String[] { "-Xdoclint:none" } );
         }
         else if ( XMVN_BUILDDEP.equals( execution ) )
         {
-- 
2.14.3

